### ISSUE TITLE
Update old docs with `deprecated` and/or `redirect` to stable, or latest. 

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#8785]
Google still have relatively high ranking for 0.10.x, and 1.x

I can build them locally, seem to work (except some qt stuff errors, but docs ok).

Though inheritance diagram are not builded and I have no clue why. 

Thoughts ? I can also at the same time upload these old docs to RTD.

**Comment by minrk** [comment]
Can you upload local builds to RTD?

**Comment by Carreau** [comment]
> Can you upload local builds to RTD? If so, we should definitely do that. I thought RTD only built from repos.

You used to be able to. Or the trick is to use a custom builder that fetch a zip and unpack it instead of actually building the stuff.

I did it a loooong time ago.

**Comment by Carreau** [comment]
Possibility, use [rel=canonical](https://yoast.com/rel-canonical/), that's what read the docs uses.

**Comment by techtonik** [comment]
So, move to RTD? Or there is a theming problem that it is not as beautiful as https://ipython.org/ipython-doc/3/index.html ? Or there is a problem to sync docs and links between sites?

**Comment by minrk** [comment]
Current docs and future docs are already on RTD. This is just a question of what to do with old versions.

**Comment by techtonik** [comment]
Try `<link rel=canonical>` ? https://en.wikipedia.org/wiki/Canonical_link_element

**Comment by takluyver** [comment]
I don't think that's quite the right thing - canonical links are for when the exact same document is available at two different URLs, and the reason we have versioned docs is that the content is not the same.

I see that the docs for versions 1, 2 and 3 now have a warning at the top that they're for an old version of IPython. I think that's about the best we can do.

**Comment by Carreau** [comment]
> Try <link rel=canonical> ? https://en.wikipedia.org/wiki/Canonical_link_element

I already proposed that a 2 messages above yours.


> I don't think that's quite the right thing - canonical links are for when the exact same document is available at two different URLs, and the reason we have versioned docs is that the content is not the same.

Read the docs does that for "stable", which is likely the version people want to be on. 
It is useful for indexing purpose.

 Google does not know how to read the warning at top of pages. Also you  can be dropped by google directly in a section and don't get a chance to see the warning. 

So I would be +1 on rel-canonical. Especially for v1,2 and 3 that have been built on RTD now.

**Comment by techtonik** [comment]
> I see that the docs for versions 1, 2 and 3 now have a warning at the top that they're for an old version of IPython. I think that's about the best we can do.

It is still an issue with 0.10 - https://www.google.com/search?q=oinspect leads to https://ipython.org/ipython-doc/rel-0.10.2/html/api/generated/IPython.OInspect.html

> I already proposed that a 2 messages above yours.

Sorry, my fault - too many signals to handle.

> Google does not know how to read the warning at top of pages.

I am pretty sure it can read and understand them. =) Baidu can do a lot more, though.

> +1 on rel-canonical - everything that moves peopleaway from outdated info helps.

If it is possible to update 0.10.2 docs we a warning, we can see if Google changes its results in a few weeks.

**Comment by Carreau** [comment]
> I am pretty sure it can read and understand them.

No not that much, version 1,2,3 docs are still (too) visible on google search. 

> If it is possible to update 0.10.2 docs with new warning, we can see if Google changes its results in a few weeks.

The all **point** of this issue is to update old docs. And as stated in the first comment of this issue : getting them to rebuild completely is hard, as **some part** don't rebuild. 

So if you figure out how to rebuild them successfully, that would be great. If you figure out how to rebuild on RTD it's even better. 

The last point is to also find time to do it.

**Comment by techtonik** [comment]
> No not that much, version 1,2,3 docs are still (too) visible on google search.

Have a prooflink to check?

> The all point of this issue is to update old docs. And as stated in the first comment of this issue : getting them to rebuild completely is hard, as some part don't rebuild.

Do you have logs for that and repeatable sequence of commands?

> Though inheritance diagram are not builded and I have no clue why.

Maybe a plugin or Graphviz is missing. Also, 0.10.x was generated with Sphinx 1.1pre, which itself may be buggy.

As a last resort, download and patch HTML.

**Comment by Carreau** [comment]
> Have a prooflink to check?

This google search for me:

<img width="770" alt="screen shot 2016-01-28 at 14 08 03" src="https://cloud.githubusercontent.com/assets/335567/12660477/ca5b03fc-c5c8-11e5-946e-935b0084562a.png">

> Do you have logs for that and repeatable sequence of commands?

at least for 0.11
$ python setup.py install 
$ git checkout rel-0.11
$ cd docs
$ make html

it **should** clone [this repo](https://github.com/ipython/ipython-doc/tree/gh-pages) in `docs/gh-pages`, 
which should make things easy to compare. 

> Maybe a plugin or Graphviz is missing. Also, 0.10.x was generated with Sphinx 1.1pre, which itself may be buggy.

It's fine if for 1 build we have to revert to sphinx 1.1pre, even if you make just a small progress on that, more instruction on this issue would be welcome, people could take over later.

**Comment by techtonik** [comment]
```
pip install sphinx pygments
git clone https://github.com/ipython/ipython.git -b 0.10.2
cd ipython
python setup.py install
cd docs
make html
```

```
Extension error:
Could not import extension inheritance_diagram (exception: cannot import name xfileref_role)
make: *** [html] Error 1
```
Seems pretty clear.

**Comment by techtonik** [comment]
The solution seems here https://github.com/nilmtk/nilmtk/issues/87#issuecomment-31749540

**Comment by techtonik** [comment]
Here is the fix that was not backported to 0.10 https://github.com/takluyver/ipython/commit/f657f2b2a6e723c8633e55155fa54f9c3a3050f5

**Comment by Carreau** [comment]
Hum, yes, the build get a bit further with this fix, that's a step forward. 

though now we need `twisted`, to be able to import `IPython.frontend.asyncfrontendbase`. 
even with twisted, install it's a bit hard to get the import:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)

/Users/bussonniermatthias/dev/ipython/IPython/frontend/<ipython console> in <module>()

/Users/bussonniermatthias/dev/ipython/IPython/frontend/asyncfrontendbase.py in <module>()
     28                                            IFrontEndFactory)
     29 from IPython.kernel.core.history import FrontEndHistory
---> 30 from IPython.kernel.engineservice import IEngineCore
     31
     32 #-----------------------------------------------------------------------------

/Users/bussonniermatthias/dev/ipython/IPython/kernel/engineservice.py in <module>()
    337         del _apiDict[id]
    338
--> 339 class EngineService(object, service.Service):
    340     """Adapt a IPython shell into a IEngine implementing Twisted Service."""
    341

/Users/bussonniermatthias/anaconda3/envs/27/lib/python2.7/site-packages/zope/interface/advice.pyc in advise(name, bases, cdict)
    139             meta = determineMetaclass(bases, previousMetaclass)
    140
--> 141         newClass = meta(name,bases,cdict)
    142
    143         # this lets the callback replace the class completely, if it wants to

TypeError: Error when calling the metaclass bases
    Cannot create a consistent method resolution
order (MRO) for bases Service, object
```

**Comment by techtonik** [comment]
I can only guess that `zope.interface` needs to be downgraded to the version that was available at that time. And try `sphinx==1.1` too.

My full LXD setup to get clean state checkout with a fresh Ubuntu:
```
lxc launch ilo:ubuntu/wily/amd64 isphinx
lxc exec isphinx -- apt-get install -y git python python-pip
lxc exec isphinx -- git clone https://github.com/ipython/ipython.git -b 0.10.2
```
Now I need to downgrade Sphinx and see what happens.

**Comment by techtonik** [comment]
Looks like 0.10.2 was released on Fri Apr 8 23:17:32 2011 -0700
At that time active versions were:

- twisted==11  - http://labs.twistedmatrix.com/2011/04/twisted-1100-released.html
- zope.interface==3.5.4  - https://pypi.python.org/pypi/zope.interface/3.6.7#id13
- sphinx<1.1   - https://pypi.python.org/pypi/Sphinx/1.1

**Comment by techtonik** [comment]
Nah, doesn't work with those. Although error message is different:
```
reading sources... [ 20%] api/generated/IPython.frontend.frontendbase

Exception occurred:
  File "/root/ipython/docs/sphinxext/docscrape.py", line 466, in __init__
    raise ValueError("Initialise using a class. Got %r" % cls)
ValueError: Initialise using a class. Got <InterfaceClass IPython.frontend.frontendbase.IFrontEnd>
```
